devenv/sync.py

```diff
@@ -112,8 +112,39 @@ def main(context: dict[str, str]) -> int:
         reporoot,
         venv_dir,
         (
-            ("javascript dependencies", ("make", "install-js-dev")),
-            ("python dependencies", ("make", "install-py-dev")),
+            (
+                "javascript dependencies",
+                (
+                    "bash",
+                    "-euo",
+                    "pipefail",
+                    "-c",
+                    """
+NODE_ENV=development ./.devenv/bin/yarn install --frozen-lockfile
+yarn check --verify-tree || yarn install --check-files
+""",
+                ),
+            ),
+            (
+                "python dependencies",
+                (
+                    "bash",
+                    "-euo",
+                    "pipefail",
+                    "-c",
```

Comment:
imo these should not be done in a shell -- we should have python invoke each of these commands

Reply:
ah note to self i need to keep using the old `make install-py-dev` etc. if `repo` is getsentry

Reply:
on second thought it's probably just better to have getsentry have its own `sync.py` - there'll be some duplicated code but getsentry make macros are sufficiently different to warrant this i think - config.ini can stay shared

Reply:
https://github.com/getsentry/getsentry/pull/13568

Reply:
> imo these should not be done in a shell -- we should have python invoke each of these commands

hmm going back to this: run_procs is gonna have to be rewritten if we want js + py (more like, jobs) to remain parallel but serial execution of procs (more like, tasks) inside - can i leave this for a followup since the main scope here is to just move the make stuff?

Reply:
yeah I suspect `run_procs` is not the right abstraction we want here -- and a plain-ole `ThreadPoolExecutor` / `ProcessPoolExecutor` is probably going to do better